### PR TITLE
Newchat

### PR DESCRIPTION
**Pull request by alegomez1**
Set up of the chat functionality

### CODE REVIEW

File: models/ChatBox.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require('mongoose')
```
Comment (AndresMWeber) [inline models/ChatBox.js]: We need to remove file before this commit is approved

File: public/javascripts/chat.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const socket = io()
```
Comment (AndresMWeber) [inline public/javascripts/chat.js]: Don't need this anymore.

File: public/javascripts/chat.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+const server_callback = (message) => {
+    console.log(`Server: ${message}`)
+}
+
+socket.on('update-username', (username) => {
+    socket.username = username
+})
+
+socket.on('emit-message', (message) => {
```
Comment (AndresMWeber) [inline public/javascripts/chat.js]: Not using this anymore

File: public/javascripts/chat.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    if (containerHeight - newMessageHeight <= scrollOffset) {
+        $messages.scrollTop = $messages.scrollHeight
+    }
+}
+
+const server_callback = (message) => {
+    console.log(`Server: ${message}`)
+}
+
+socket.on('update-username', (username) => {
```
Comment (AndresMWeber) [inline public/javascripts/chat.js]: Nor using this.

File: public/javascripts/chat.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            lat: position.coords.latitude,
+            lon: position.coords.longitude
+        }, (message) => {
+            $locationButton.removeAttribute('disabled')
+            server_callback(message)
+        })
+    })
+})
+
+socket.on('update-users', ({users, user}) => {
```
Comment (AndresMWeber) [inline public/javascripts/chat.js]: Remove

File: public/javascripts/chat.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+    const html = Mustache.render($sidebarTemplate, {
+        users,
+        room: user.room
+    })
+    document.querySelector('#sidebar').innerHTML = html
+})
+
+socket.emit('join', { username, room }, (message) => {
```
Comment (AndresMWeber) [inline public/javascripts/chat.js]: Remove

File: public/javascripts/messages.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+          createdAt: new Date().getTime()
+      },
+      ...extras
+  }
+  return combined
+}
+
+module.exports = {
+  generateMessage
+}
```
Comment (AndresMWeber) [inline public/javascripts/messages.js]: need a newline

File: public/javascripts/users.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
+const users = []
```
Comment (AndresMWeber) [inline public/javascripts/users.js]: Can remove this file, redundant.

File: routes/addPost.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
+const express = require('express');
```
Comment (AndresMWeber) [inline routes/addPost.js]: Remove this file

File: routes/beef.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const router = express.Router();
+
+/* GET home page */
+router.get('/beef', (req, res, next) => {
+    res.render('beef');
+});
+
+// router.get('/api-key', (req, res, next) => {
```
Comment (AndresMWeber) [inline routes/beef.js]: You can remove this instead of commenting

File: routes/login.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
```
Comment (AndresMWeber) [inline routes/login.js]: We need to decide whether we are using beef.js or index.js but definitely not both.

File: routes/signup.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const router = express.Router();
+
+/* GET home page */
+router.get('/signup', (req, res, next) => {
+    res.render('./signup-views/signup');
+});
+
+// router.get('/api-key', (req, res, next) => {
```
Comment (AndresMWeber) [inline routes/signup.js]: Remove this

File: routes/login.js [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+const router = express.Router();
+
+/* GET home page */
+router.get('/login', (req, res, next) => {
+    res.render('./login-views/login');
+});
+
+// router.get('/api-key', (req, res, next) => {
```
Comment (AndresMWeber) [inline routes/login.js]: Remove this

File: public/javascripts/script.js [hunk 14]
```diff
@@ -5,6 +5,41 @@ document.addEventListener('DOMContentLoaded', () => {
 }, false);
 
 
+//*****************Axios Code */
```
Comment (AndresMWeber) [inline public/javascripts/script.js]: Don't need this

**Comment by adriansdk** [comment]
Closing request as this was supposed to be merging with dev.

**Pull request closed without merging**